tests/phpunit/tests/formatting/excerptRemoveFootnotes.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @group formatting
+ * @ticket 58805
+ *
+ * @covers ::excerpt_remove_footnotes
+ */
+
+class Tests_Formatting_ExcerptRemoveFootnotes extends WP_UnitTestCase {
+	/**
+	 * @ticket 58805
+	 *
+	 * @dataProvider data_remove_footnotes
+	 *
+	 * @param string $expected  Expected output.
+	 * @param string $content   Content to run strip_shortcodes() on.
+	 * @param string $message   The test message.
+	 */
+	public function test_remove_footnotes( $expected, $content, $message = '' ) {
+		$this->assertSame( $expected, excerpt_remove_footnotes( $content ), $message );
+	}
+
+	public function data_remove_footnotes() {
+		return array(
+			array(
```

Comment:
Can we get descriptive keys on these datasets so it's clear at first glance what each is testing?

Such as:
```php
'no footnote'                       => array(
'one footnote'                      => array(
'multiple footnotes'                => array(
'footnotes around non-latin script' => array(
```